src/app/public/modules/radio/radio-group.component.ts

```diff
@@ -61,22 +61,16 @@ export class SkyRadioGroupComponent implements AfterContentInit, ControlValueAcc
 
   @Input()
   public set value(value: any) {
-    const currentValue = this._value;
-    if (currentValue !== value) {
+    // The null check is needed to address a bug in Angular 4.
+    // writeValue is being called twice, first time with a phantom null value
+    // See: https://github.com/angular/angular/issues/14988
+    // tslint:disable-next-line:no-null-keyword
+    const isNewValue = value !== this._value && value !== null;
+
+    if (isNewValue) {
       this._value = value;
+      this.onChange(this._value);
       this.updateCheckedRadioFromValue();
-
-      // Explicitly check both `undefined` and `null` (Angular's empty value) so that
-      // we can support boolean values for the form control.
-      /* tslint:disable:no-null-keyword */
-      if (
```

Comment:
I feel like these were here in order to make sure that the initial value doesn't set the `touched` or `dirty` flags. I might just be missing it but is there a test to ensure that we aren't accidentally setting those flags when an initial value is given? I see a test for when there isn't a value given but not for when there is.

Reply:
You're right to look for that... Steve and I actually discussed that earlier. So, there's a known issue here with those states not being set correctly. It was already present in the code before this PR. We decided to work on that separately after.

As far as this particular code block goes, it was to ensure we were allowing boolean values. Steve made a PR to fix that a few weeks ago, and I confirmed that my changes still uphold the intent of the fix. While there was a test to look at the form control flags, it didn't necessarily cover the scenario where the initial value is set. I think that's a good idea, so I added it! Thanks for bringing that up.

Reply:
Thanks for adding that test!